numpy/_core/src/multiarray/stringdtype/static_string.c

```diff
@@ -552,15 +522,6 @@ NpyString_newemptysize(size_t size, npy_packed_static_string *out,
 
     _npy_static_string_u *out_u = (_npy_static_string_u *)out;
 
-    unsigned char flags =
-            out_u->direct_buffer.size_and_flags & ~NPY_SHORT_STRING_SIZE_MASK;
-
-    if (size == 0) {
```

Comment:
I'm still wondering about special-casing `size == 0` for a string that is already in the arena, to just set the size to 0, but keep the offset, so that, like missing, it can be recovered (unlike a short string). But probably not worth it (and could be done in follow-up anyway).